create ijepa modelcard (ref : PR  #36979 ).

dhruvmalik007 opened the pull request:
# What does this PR do?

<!--
Congratulations! You've made it this far! You're not quite done yet though.

Once merged, your PR is going to appear in the release notes with the title you set, so make sure it's a great title that fully reflects the extent of your awesome contribution.

Then, please replace this with a description of the change and which issue is fixed (if applicable). Please also include relevant motivation and context. List any dependencies (if any) that are required for this change.

Once you're done, someone will review your PR shortly (see the section "Who can review?" below to tag some potential reviewers). They may suggest changes to make the code even better. If no one reviewed your PR after a week has passed, don't hesitate to post a new comment @-mentioning the same persons---sometimes notifications get lost.
-->

<!-- Remove if not applicable -->

#36979 : adds the descriptive model card based on the latest standard.


## Before submitting
- [X] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case
- [X] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?


## Who can review?

@stevhliu

dhruvmalik007 committed:
wip: adding first version of the IJEPA model card.

dhruvmalik007 committed:
Merge branch 'huggingface:main' into dhruvmalik007/create-ijepa-modelcard

stevhliu commented on docs/source/en/model_doc/ijepa.md:
````diff
@@ -66,14 +77,33 @@ def infer(image):
+    outputs = model(**inputs)
+    ```
+
 ## Resources
````
Can remove the `## Resources` section as well

stevhliu commented on docs/source/en/model_doc/ijepa.md:
```diff
@@ -14,36 +14,47 @@ rendered properly in your Markdown viewer.
-This paper demonstrates an approach for learning highly semantic image representations without relying on hand-crafted data-augmentations. We introduce the Image- based Joint-Embedding Predictive Architecture (I-JEPA), a non-generative approach for self-supervised learning from images. The idea behind I-JEPA is simple: from a single context block, predict the representations of various target blo ...
+The example below demonstrates how to extract image features with [`Pipeline`] or the [`AutoModel`] class.
 
-<img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/model_doc/ijepa_architecture.jpg"
```
Lets keep the image

stevhliu reviewed (commented):
Thanks!

dhruvmalik007 committed:
refactor based on the @stevhliu feedbacks

dhruvmalik007 committed:
Merge branch 'huggingface:main' into dhruvmalik007/create-ijepa-modelcard

dhruvmalik007 committed:
Merge branch 'dhruvmalik007/create-ijepa-modelcard' of https://github.com/Frontier-tech-consulting/transformers into local changes

stevhliu commented on docs/source/en/model_doc/ijepa.md:
```diff
@@ -14,85 +14,121 @@ rendered properly in your Markdown viewer.
+## Quantization
 
-## IJepaConfig
+The following example demonstrates how to quantize I-JEPA models using bitsandbytes for 4-bit and 8-bit quantization to reduce model size and memory usage while maintaining performance. More quantization methods are discussed [here](https://huggingface.co/docs/transformers/v4.53.2/quantization/overview).
```
```suggestion
Quantization reduces the memory burden of large models by representing the weights in a lower precision. Refer to the [Quantization](../quantization/overview) overview for more available quantization backends.

The example below uses [bitsandbytes](../quantization/bitsandbytes) to only quantize the weights to 4-bits.
```

stevhliu commented on docs/source/en/model_doc/ijepa.md:
```diff
@@ -14,85 +14,121 @@ rendered properly in your Markdown viewer.
+The following example demonstrates how to quantize I-JEPA models using bitsandbytes for 4-bit and 8-bit quantization to reduce model size and memory usage while maintaining performance. More quantization methods are discussed [here](https://huggingface.co/docs/transformers/v4.53.2/quantization/overview).
 
-[[autodoc]] IJepaConfig
+<hfoptions>
+<hfoption id="Quantization">
```
```suggestion
```

stevhliu commented on docs/source/en/model_doc/ijepa.md:
```diff
@@ -14,85 +14,121 @@ rendered properly in your Markdown viewer.
-[[autodoc]] IJepaForImageClassification
-    - forward
\ No newline at end of file
+</hfoption>
+</hfoptions>
```
```suggestion
```

stevhliu commented on docs/source/en/model_doc/ijepa.md:
````diff
@@ -14,85 +14,121 @@ rendered properly in your Markdown viewer.
-
-[[autodoc]] IJepaModel
-    - forward
+```py
````
```py
import torch
from transformers import BitsAndBytesConfig, AutoModel, AutoProcessor
from datasets import load_dataset

quantization_config = BitsAndBytesConfig(
    load_in_4bit=True,
    bnb_4bit_quant_type="nf4",
    bnb_4bit_compute_dtype=torch.bfloat16,
    bnb_4bit_use_double_quant=True,
)

url_1 = "http://images.cocodataset.org/val2017/000000039769.jpg"
url_2 = "http://images.cocodataset.org/val2017/000000219578.jpg"
image_1 = Image.open(requests.get(url_1, stream=True).raw)
image_2 = Image.open(requests.get(url_2, stream=True).raw)

processor = AutoProcessor.from_pretrained("facebook/ijepa_vith14_1k")
model = AutoModel.from_pretrained("facebook/ijepa_vith14_1k", quantization_config=quantization_config, torch_dtype="auto", attn_implementation="sdpa")


def infer(image):
    inputs = processor(image, return_tensors="pt")
    outputs = model(**inputs)
    return outputs.last_hidden_state.mean(dim=1)


embed_1 = infer(image_1)
embed_2 = infer(image_2)

similarity = cosine_similarity(embed_1, embed_2)
print(similarity)
```

stevhliu reviewed (commented):
Two more things:

- add the API references back since they shouldn't be removed (`[[autodoc]]`)
- remember to add the image back

dhruvmalik007 commented:
> Two more things:
> 
> * add the API references back since they shouldn't be removed (`[[autodoc]]`)
> * remember to add the image back

thanks for reminding the  image.  this was indeed skipped by me .

dhruvmalik007 committed:
refactor:
- revert the accidental removal of the autodoc api description and the image reerece architecture

- general context updation.

stevhliu reviewed (approved):
Thanks, just a few more things!

dhruvmalik007 committed:
- changes of model for example quantization.
- merging the  quantization content.
